mmdet/core/results/results.py

```diff
@@ -0,0 +1,441 @@
+import copy
+import itertools
+from itertools import chain
+
+import numpy as np
+import torch
+
+from mmdet.utils.util_mixins import NiceRepr
+
+
+class Results(NiceRepr):
+    """A general data structure to store the model's results.
+
+    The attributes of `Results` are divided into two parts,
+    the `meta_info_field` and the `results_field` respectively.
+
+    The `meta_info_field` usually includes the information about
+    the image such as filename, image_shape, padding_shape, etc.
+    All attributed in this filed is unmodifiable once set,
+    but the user can add new meta information with
+    `add_meta_info` function.
+
+
+    The model predictions are stored in `results_field`.
+    Models predictions can be accessed
+    or modified by dict-like or object-like operations
+    such as  `.` , `[]`, `in`, `del`, `pop(str)` `get(str)`, `keys()`,
+    `values()`, `items()`. User can also apply tensor-like methods
+    to all obj:`torch.Tensor` in the `results_filed`,
+    such as `.cuda()`, `.cpu()`, `.numpy()`, `device`, `.to()`
+    `.detach()`, `.numpy()`
+
+    Args:
+        img_meta (dict): A dict contains the meta information
+            of image. such as `img_shape`, `scale_factor`, etc.
+
+    Examples:
+        >>> from mmdet.core.results.results import Results
+        >>> img_meta = dict(img_shape=(800, 1196, 3), pad_shape=(800, 1216, 3))
+        >>> results = Results(img_meta)
+        >>> img_shape in results
+        True
+        >>> results.det_labels = torch.LongTensor([0, 1, 2, 3])
+        >>> results["det_scores"] = torch.Tensor([0.01, 0.1, 0.2, 0.3])
+        >>> print(results)
+        <Results(
+
+          META INFORMATION
+        img_shape: (800, 1196, 3)
+        pad_shape: (800, 1216, 3)
+
+           PREDICTIONS
+        shape of det_labels: torch.Size([4])
+        shape of det_scores: torch.Size([4])
+
+        ) at 0x7f84acd10f90>
+        >>> resutls.det_scores
+        tensor([0.0100, 0.1000, 0.2000, 0.3000])
+        >>> results.det_labels
+        tensor([0, 1, 2, 3])
+        >>> results['det_labels']
+        tensor([0, 1, 2, 3])
+        >>> 'det_labels' in results
+        True
+        >>> results.img_shape
+        (800, 1196, 3)
+        >>> 'det_scores' in results
+        True
+        >>> del results.det_scores
+        >>> 'det_scores' in results
+        False
+        >>> det_labels = results.pop('det_labels', None)
+        >>> det_labels
+        tensor([0, 1, 2, 3])
+        >>> 'det_labels' in results
+        >>> False
+    """
+
+    def __init__(self, img_meta=None):
```

Comment:
for naming issue, here it is not necessarily to be img_meta, we can simply use meta for more general usage